doctr/models/predictor/tensorflow.py

```diff
@@ -43,13 +42,22 @@ def __init__(
     def __call__(
         self,
         pages: List[Union[np.ndarray, tf.Tensor]],
+        rotate_document: bool = False,
         **kwargs: Any,
     ) -> Document:
 
         # Dimension check
         if any(page.ndim != 3 for page in pages):
             raise ValueError("incorrect input shape: all pages are expected to be multi-channel 2D images.")
 
+        # Detect document rotation and rotate pages
+        if rotate_document:
+            page_angles = []
+            for i, page in enumerate(pages):
+                page_angle = estimate_orientation(page)
+                page_angles.append(page_angle)
+                pages[i] = rotate_page(page, page_angle)
```

Comment:
Let's do the same with list comprehensions :) 

```python
page_orientations = [estimate_orientation(page) for page in pages]
pages = [rotate_page(page, angle) for page, angle in zip(pages, page_orientations)]
```

Reply:
sure !